src/event_message/parse.rs

```diff
@@ -6,23 +6,23 @@ use crate::{
 use nom::{branch::*, combinator::*, error::ErrorKind, multi::*, sequence::*};
 use serde_transcode::transcode;
 
-fn json_message(s: &str) -> nom::IResult<&str, EventMessage> {
-    let mut stream = serde_json::Deserializer::from_slice(s.as_bytes()).into_iter::<EventMessage>();
+fn json_message(s: &[u8]) -> nom::IResult<&[u8], EventMessage> {
+    let mut stream = serde_json::Deserializer::from_slice(s).into_iter::<EventMessage>();
     match stream.next() {
         Some(Ok(event)) => Ok((&s[stream.byte_offset()..], event)),
         _ => Err(nom::Err::Error((s, ErrorKind::IsNot))),
     }
 }
 
-fn cbor_message(s: &str) -> nom::IResult<&str, EventMessage> {
-    let mut stream = serde_cbor::Deserializer::from_slice(s.as_bytes()).into_iter::<EventMessage>();
+fn cbor_message(s: &[u8]) -> nom::IResult<&[u8], EventMessage> {
+    let mut stream = serde_cbor::Deserializer::from_slice(s).into_iter::<EventMessage>();
     match stream.next() {
         Some(Ok(event)) => Ok((&s[stream.byte_offset()..], event)),
         _ => Err(nom::Err::Error((s, ErrorKind::IsNot))),
     }
 }
 
-fn message(s: &str) -> nom::IResult<&str, EventMessage> {
+pub fn message(s: &[u8]) -> nom::IResult<&[u8], EventMessage> {
```

Comment:
Is this intentional to expose that that function, does not seems to be used in that commit.

Reply:
you're right, thats leftover from the branch I originally did this change in